File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;

        for (string& str : strs){
            string count(26,0);

            for (char x : str){
                count [x - 'a']++;
            }
            groups[count].push_back(str);
        }
        vector<vector<string>> result;
        for (auto& group : groups) {
            result.push_back(group.second);
        }

        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams_collection = defaultdict(list)

        for s in strs: 
            s_sorted = "".join(sorted(s))
            anagrams_collection[s_sorted].append(s)
        return list(anagrams_collection.values())